### DISCUSSION TITLE
`vim` keyword is defined in  nvim config but not in `my/local/plugin` dir.

### DISCUSSION THREAD

**Discussion by molleweide** [neovim/neovim#30152] [Q&A] [answered]
Hi,

I dunno if this is a stupid question or not..

Lua_ls is fine with `vim` keyword inside nvim config but complains about `vim` keyword in local plugin.

Is everything working as expected here or do I need to add additional config for local plugins/workspaces?

**Comment by telemachus** [comment, marked as answer]
It's not a stupid question. You can configure the language sever to recognize specific globals, but in this case, even better is to teach the language server about `vim.env.VIMRUNTIME`.

You can see how nvim-lspconfig recommends here: https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#lua_ls

There's (lots) more information about possible configuration for the Lua language server [on its wiki](https://luals.github.io/wiki/settings).

**Reply by molleweide** [reply]
Allright, thanks.

Adding diagnostics.globals did the trick it seems.

I thought that the default lspconfig would already handle this problem but there seems to be more to it than I could imagine initially.
